File: app/src/main/java/com/example/myapplication/domain/repository/UserRepository.java
package com.example.myapplication.domain.repository;

import com.example.myapplication.domain.model.User;

import java.util.List;

public interface UserRepository {
    boolean bindService();

    void unbindService();

    void addConnectionStateListener(ConnectionStateListener listener);

    void removeConnectionStateListener(ConnectionStateListener listener);

    List<User> getUsers();

    void addUser(User user);

    void updateUser(User user);

    void deleteUser(long userId);
}

File: app/src/main/java/com/example/myapplication/domain/repository/UserRepository.kt
package com.example.myapplication.domain.repository

import com.example.myapplication.domain.model.User

interface UserRepository {
    fun bindService(): Boolean

    fun unbindService()

    fun addConnectionStateListener(listener: ConnectionStateListener)

    fun removeConnectionStateListener(listener: ConnectionStateListener)

    fun getUsers(): List<User>

    fun addUser(user: User)

    fun updateUser(user: User)

    fun deleteUser(userId: Long)
}
